utils/scale-test

```diff
@@ -101,23 +117,52 @@ def run_many(args):
 
     ast = gyb.parse_template(args.file.name, args.file.read())
     rng = range(args.begin, args.end, args.step)
-    if args.multi_file:
-        return (rng, [run_once(args, ast, rng[0:i+1]) for i in range(len(rng))])
+    if args.multi_file or args.sum_multi:
+        return (rng, [run_once(args, ast, range(i)) for i in rng])
     else:
         return (rng, [run_once(args, ast, [r]) for r in rng])
 
 
+def linear_regression(x, y):
+   # By the book: https://en.wikipedia.org/wiki/Simple_linear_regression
+   n = len(x)
+   if n == 0:
+       return 0, 0
+   prod_sum = 0
+   sum_x = 0
+   sum_y = 0
+   sum_prod = 0
+   sum_x_sq = 0
+   for i in range(n):
```

Comment:
This may be simpler as

```swift
for a, b in zip(x, y):
    sum_x += a
    sum_y += b 
    sum_prod += a * b
    sum_x_sq += a * a
```

Or even just using `sum(x)`, `sum(y)`, `sum(a * b for a, b in zip(x, y))`, since I can't imagine there'll be enough data for doing multiple passes to matter.

As we discussed before, the subtractions below may be subject to catastrophic cancellation/numerical instability,  but I think this shouldn't be too bad: it's worst when the numbers are all similar and/or there's a lot of them, but the nature of this script should mean neither of those are true.